R/1.0_module_data_summary.R

```diff
@@ -140,16 +140,24 @@ get_filter_overview <- function(teal_data) {
     simplify = FALSE
   )
 
-  rows <- lapply(
+  child_parent <- sapply(
     datanames,
+    function(i) teal.data::parent(joinkeys, i),
+    USE.NAMES = TRUE,
+    simplify = FALSE
+  )
+  ordered_datanames <- topological_sort(child_parent)
+  ordered_datanames <- intersect(ordered_datanames, datanames)
+
+  rows <- lapply(
+    ordered_datanames,
```

Comment:
I'd apply this to `teal_data_datanames` function so that we keep this order consistent in whole `teal` application.

Reply:
@gogonzo do you mean `teal.data::datanames()` ? https://github.com/insightsengineering/teal.data/blob/main/R/teal_data-datanames.R#L34

Sure, that would make sense. So each time `datanames(teal_data)` is called we already get names in the topological sort order. So let's abandon this branch and I'll provide a PR in `teal.data`

Reply:
I mean `teal:::teal_data_datanames`

Reply:
Created a small PR in `teal.data` package. Still WIP. Asked @averissimo for the review https://github.com/insightsengineering/teal.data/pull/318

Reply:
ignore my PR, I will investigate `teal:::teal_data_datanames`